### PR TITLE
Make passive gates great

### PR DESCRIPTION
**Pull request by Partmedia**
## About the PR 
Passive gates are currently not very useful. They are mapped one time in one map (the Syndie Puddle). They feature low transfer rates and no added benefit.

This patch gives passive gates two important features:

1. The passive gate equalizes pressure between the input and output pipe nets instantly, which means that they are nearly equivalent to connecting two pipe nets and are much faster than pressure and volume pumps. They are still one-way and can only move gas from high pressure to low pressure, though.

2. Passive gates have a built-in flow meter. This means that you can install them in a pipe net and measure flow, i.e. to determine whether or not there is a giant leak on the station.

**Screenshots**

Pressure gates now equalize input and output pressure instantly:
passive-gate-1

This atmos tech did not increase the pressure of the pressure pumps going into the mixer, and so is pumping up dist very slowly. Bad atmos tech!
passive-gate-2

This based atmos tech has configured the pressure pumps going into the mixer so that dist can be pumped up quickly. They can check the result of their handiwork on the flowmeter, which shows much more flow than the un-based atmos tech.
passive-gate-3

**Changelog**

:cl: notafet
- Tweak: Passive gates now equalize pressure instantly and come with a built-in gas flow meter.

### CODE REVIEW

**Comment by Emisse** [comment]
thank you this looks very useful

File: Content.Server/Atmos/Piping/Binary/EntitySystems/GasPassiveGateSystem.cs [hunk 1]
```diff
@@ -12,43 +15,91 @@ namespace Content.Server.Atmos.Piping.Binary.EntitySystems
...
                 _atmosphereSystem.Merge(outlet.Air, inlet.Air.Remove(transferMoles));
             }
+
+            // Update transfer rate with an exponential moving average.
+            var tau = 1;    // Time constant (averaging time) in seconds
+            var a = dt/tau;
+            gate.FlowRate = a*dV/tau + (1-a)*gate.FlowRate; // in L/sec
+        }
+
+        private void OnGetExamineVerbs(EntityUid uid, GasPassiveGateComponent component, GetVerbsEvent<ExamineVerb> args)
```
Comment (Elijahrane) [inline Content.Server/Atmos/Piping/Binary/EntitySystems/GasPassiveGateSystem.cs:83]: This should probably just use ExaminedEvent

File: Resources/Locale/en-US/atmos/gas-passive-gate-component.ftl [hunk 2]
```diff
@@ -0,0 +1,2 @@
+gas-passive-gate-component-tooltip = Check the flow rate
+gas-passive-gate-examined = Flow Rate: {$flowRate} L/sec
```
Comment (Elijahrane) [inline Resources/Locale/en-US/atmos/gas-passive-gate-component.ftl:2]: Make the number colored

**Review by Elijahrane — CHANGES_REQUESTED** [review]

File: Content.Server/Atmos/Piping/Binary/EntitySystems/GasPassiveGateSystem.cs (hunk 1 above)
Comment (Partmedia) [reply Content.Server/Atmos/Piping/Binary/EntitySystems/GasPassiveGateSystem.cs:83]: I'll make the change.

File: Resources/Locale/en-US/atmos/gas-passive-gate-component.ftl (hunk 2 above)
Comment (Partmedia) [reply Resources/Locale/en-US/atmos/gas-passive-gate-component.ftl:2]: Suggested color?

Comment (Elijahrane) [reply Resources/Locale/en-US/atmos/gas-passive-gate-component.ftl:1]: Some sort of blue is pretty neutral

**Comment by Partmedia** [comment]
Update examine text:
passive-gate-new-1

**Review by Elijahrane — APPROVED** [review]

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**